web/jobs/jobs.go

```diff
@@ -495,19 +495,21 @@ func purgeJobs(c echo.Context) error {
 		workers = strings.Split(workersParam, ",")
 	}
 
+	allJobs, err := job.GetAllJobs(instance)
+	if err != nil {
+		return err
+	}
+
 	// Step 1: We want to get all the jobs prior to the date parameter.
 	// Jobs returned are the ones we want to remove
 	d := time.Now().Add(-dur)
-	jobsBeforeDate, err := job.GetJobsBeforeDate(instance, d)
-	if err != nil && err != job.ErrNotFoundJob {
-		return err
-	}
+	jobsBeforeDate := job.FilterJobsBeforeDate(allJobs, d)
```

Comment:
I think there is a subtle bug:

- `job.FilterJobsBeforeDate` is modifying some of the jobs in `allJobs` because of the trick `b := jobs[:0]` to use the same underlying storage for both `allJobs` and `jobsBeforeDate` (alias `b`)
- but we are also using `allJobs` later (https://github.com/cozy/cozy-stack/pull/1916/files#diff-5b7c95bb32fcec71d666c8ab794c5371R512)